conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,537 @@
+            for root, dirnames, filenames in os.walk(d):
+                for filename in filenames:
+                    filename = os.path.join(root, filename)
+                    archname = os.path.relpath(filename, d)
+                    if os.path.islink(filename):
+                        target = os.readlink(filename)
+                        f.writestr(symlink_zipinfo(archname), target)
+                    else:
+                        f.write(filename, archname)
+                for dirname in dirnames:
+                    filename = os.path.join(root, dirname)
+                    archname = os.path.relpath(filename, d)
+                    if os.path.islink(filename):
+                        target = os.readlink(filename)
+                        f.writestr(symlink_zipinfo(archname), target)
+                    else:
+                        f.write(filename, archname)
+            f.close()
+
+    def decompress(self, d):
+        with zipfile.ZipFile(file='test' + self.ext, mode='r') as f:
+            for zi in f.infolist():
+                # type 4 bits are for file type, 0xA is for S_IFLNK (symbolic link)
+                if (zi.external_attr >> 28) == 0xA:
+                    f.extract(zi, path=d)
+                    dst_name = os.path.join(d, zi.filename)
+                    data = load(dst_name)
+                    os.unlink(dst_name)
+                    os.symlink(data, dst_name)
+                else:
+                    f.extract(zi, path=d)
+                    dt = time.mktime(zi.date_time + (0, 0, -1))
+                    os.utime(os.path.join(d, zi.filename), (dt, dt))
+                    if os.name == 'posix':
+                        permissions = zi.external_attr >> 16
+                        os.chmod(os.path.join(d, zi.filename), permissions)
+
+
+class TarCompressor(Compressor):
+    read_mode = "r"
+    write_mode = "w"
+    ext = ".tar"
+
+    def compress(self, d):
+        with tarfile.open(name='test' + self.ext, mode=self.write_mode, format=tarfile.GNU_FORMAT) as f:
+            inodes = dict()
+            for root, dirnames, filenames in os.walk(d):
+                for filename in filenames:
+                    filename = os.path.join(root, filename)
+                    archname = os.path.relpath(filename, d)
+
+                    ti = tarfile.TarInfo(archname)
+                    if os.path.islink(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+                    else:
+                        st = os.stat(filename)
+                        if st.st_ino in inodes:
+                            ti.type = tarfile.LNKTYPE
+                            ti.linkname = inodes[st.st_ino]
+                            ti.size = 0
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
```

Comment:
We had tons of issues with these timestamps (cc @lasote), and what happens when you try to build sources with timestamps that are probably not aligned with the current machine time.

The solution we had to implement was:
- Do not include timestamps either in ther tar or in the gz.
- Touch the files once uncompressed

I think it is necessary to investigate this well before adding back the time to the zipped artifacts.

Reply:
I am not sure what else to investigate here. we already know timestamps are very unreliable, for several reasons:
- `zip` format resolution is only 2 seconds, thus it already loses some information
- timestamp resolution is file-system dependent (e.g. NTFS has 100-nanosecond units), while `tar` provides resolution only to 1 second
- epoch start is also file-system dependent (e.g. NTFS starts at year 1601), while `tar` epoch starts at 1970 (`zip` starts at 1980)
- `zip` timestamp is 32-bit and subject to so-called problem 2038
- `tar` and `zip` only store modification time, creation and access timestamps are unconditionally lost

you may `touch` after the uncompressing (as I do in the tests), but information is already lost at that point.
probably, if you want to save timestamps reliably, you better store them to the external file (manifest?).
overwise, I'd say - just don't store them at all.